bot/exts/evergreen/youtube.py

```diff
@@ -0,0 +1,151 @@
+import logging
+from dataclasses import dataclass
+from html import unescape
+from typing import List, Optional
+from urllib.parse import quote_plus
+
+from discord import Embed
+from discord.ext import commands
+from discord.utils import escape_markdown
+
+from bot.constants import Colours, Emojis, Tokens
+
+log = logging.getLogger(__name__)
+
+KEY = Tokens.youtube
+SEARCH_API = "https://www.googleapis.com/youtube/v3/search"
+STATS_API = "https://www.googleapis.com/youtube/v3/videos"
+YOUTUBE_VIDEO_URL = "https://www.youtube.com/watch?v={id}"
+YOUTUBE_SEARCH_URL = "https://www.youtube.com/results?search_query={search}"
+RESULT = (
+    "**{index}. [{title}]({url})**\n"
+    "{post_detail_emoji} {user_emoji} {username} {view_emoji} {view_count} {like_emoji} {like_count}\n"
+)
+
+
+@dataclass
+class VideoStatistics:
+    """Represents YouTube video statistics."""
+
+    view_count: int
+    like_count: int
+
+
+@dataclass
+class Video:
+    """Represents a video search result."""
+
+    title: str
+    username: str
+    id: str
+    video_statistics: VideoStatistics
+
+
+class YouTubeSearch(commands.Cog):
+    """Sends the top 5 results of a query from YouTube."""
+
+    def __init__(self, bot: commands.Bot):
+        self.bot = bot
+
+    async def format_search_result(self, index: int, result: List[Video]) -> str:
+        """Formats search result to put in embed."""
+        return RESULT.format(
+            index=index,
+            title=result.title,
+            url=YOUTUBE_VIDEO_URL.format(id=result.id),
+            post_detail_emoji=Emojis.post_detail,
+            user_emoji=Emojis.user,
+            username=result.username,
+            view_emoji=Emojis.view,
+            view_count=result.video_statistics.view_count,
+            like_emoji=Emojis.like,
+            like_count=result.video_statistics.like_count,
+        )
+
+    async def get_statistics(self, id: str) -> Optional[VideoStatistics]:
+        """Queries API for statistics of one video."""
+        async with self.bot.http_session.get(
+            STATS_API,
+            params={"part": "statistics", "id": id, "key": KEY},
+        ) as response:
+            if response.status != 200:
```

Comment:
As discussed in discord, maybe such errors should be left to a top-level handler

Reply:
@decorator-factory As it's been a bit since I've worked on this, were we thinking about making the request, and if it fails, just tell the user "Something went wrong!"